src/App.tsx

```diff
@@ -75,15 +103,25 @@ const App: React.FC<{}> = () => {
         <ToasterProvider autoDismiss>
           <Web3Provider
             tokensToWatch={tokens}
-            networkIds={[5]}
+            networkIds={[networkId]}
             onboardConfig={{
               dappId: process.env.REACT_APP_BLOCKNATIVE_DAPP_ID,
               walletSelect: {
-                wallets: [{ walletName: "metamask", preferred: true }],
+                wallets: [
+                  { walletName: "metamask", preferred: true },
+                  {
+                    walletName: "walletConnect",
+                    preferred: true,
+                    rpc,
+                  },
+                ],
               },
               subscriptions: {
-                network: (network) =>
-                  network && console.log("chainId: ", network),
+                network: (newNetworkId) => {
+                  const supported = chainbridgeConfig.chains.find(chain => chain.networkId === newNetworkId);
+                  if(supported) setNetworkId(newNetworkId);
```

Comment:
We don't have an `else` case here.

Reply:
Fixed formatting

Reply:
2. In the next line you have a log about changing the networkId. If it is not supported, the log will mislead. To avoid this: 
    a. Move current log to the scope of the `true` branch
    b. Add new log to the `false` branch.

Reply:
Updated log format